package/src/utils/utils.ts

```diff
@@ -275,3 +275,63 @@ export const getDurationLabelFromDuration = (duration: number) => {
 export function escapeRegExp(text: string) {
   return text.replace(/[-[\]{}()*+?.,/\\^$|#]/g, '\\$&');
 }
+
+/**
+ * Utility to find the index of a message in the messages array by id.
+ * @param messages
+ * @param targetId
+ * @returns number
+ */
+export const findInMessagesById = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  messages: ChannelState<StreamChatGenerics>['messages'],
+  targetId: string,
+) => {
+  const idx = messages.findIndex((message) => message.id === targetId);
+  return idx;
+};
+
+/**
+ * Utility to find the index of a message in the messages array by date.
+ * @param messages
+ * @param targetDate
+ * @returns an object with the index and the message object
+ */
+export const findInMessagesByDate = <
```

Comment:
I know we talked about this but I sort of forgot, what's the reason we're not using the baked in binary search from the LLC but have a copy of it here ?

Reply:
LLC doesn't have any method of binary search for dates that checks for date lying between two neighboring timestamps hence this was created. The clients binary search method checks for the exact date or the index of the nearest greater date.

Reply:
Sure - but why not add it ?

Reply:
Client won't be using this method anywhere in the code but will just export it, so it will be just a export from LLC. I am not sure if it makes sense in that case.

Reply:
As discussed on a call, we consider this out of scope for now